### PR TITLE
Store the last cluster on which the application was running as a DRPC annotation

### PR DESCRIPTION
**Pull request by BenamarMk**
Fixes issue: [912](https://github.com/RamenDR/ramen/issues/912)

### CODE REVIEW

File: controllers/drplacementcontrol.go [hunk 1]
```diff
@@ -30,6 +30,9 @@ const (
 	// Annotations for MW and PlacementRule
 	DRPCNameAnnotation      = "drplacementcontrol.ramendr.openshift.io/drpc-name"
 	DRPCNamespaceAnnotation = "drplacementcontrol.ramendr.openshift.io/drpc-namespace"
+
+	// Annotation for the last cluster on which the application was running
+	LastAppDeploymentCluster = "drplacementcontrol.ramendr.openshift.io/last-app-deployment-cluster"
```
Comment (nirs) [inline controllers/drplacementcontrol.go:35]: The name is not consistent with other annotations above but I guess the reason is
DRPCName*Annotation is annotation for ManifestWork and PlacementRule (and Placement?) and this annotation is for the drpc resource. Maybe having a separate `const ()` with a comment for drpc annotations will make this more clear.

Regarding the new annotation name - how it is related to preferredCluster and failoverCluster?

```
  spec:
    failoverCluster: dr2
    preferredCluster: dr1
    ...
```

Isn't it the same as preferredCluster during failover, and failoverCluster during relocate?

Regardless, can we use something more consistent with these names, like last-cluster? or last-primary-cluster?

**Review by nirs — COMMENTED** [review]
Can you explain in the commit message why we need to keep the last cluster and how we plan to use it?

Also, how the value will change during the main flows: deploy, failover, relocate?

File: controllers/drplacementcontrol.go (hunk 1 above)
Comment (BenamarMk) [reply controllers/drplacementcontrol.go:35]: The name was proposed by the UX team, and I find it quite logical. Essentially, `last-app-deployment-cluster` refers to the cluster where the app was last deployed on.

**Review by ShyamsundarR — APPROVED** [review]
I was thinking of discussing more hub recovery cases and what we can do in these instances. We can reserve that for the next hub-recovery improvements phase though. merging soon.

**Pull request merged**